storage.bs

```diff
@@ -94,37 +100,251 @@ function retrieveNextChunk(nextChunkInfo) {
+ <a>storage bucket</a>.
+
+ <li><p>Otherwise, if <var>type</var> is "<code>session-storage</code>", set <var>bucket</var> to a
+ new <a>session storage bucket</a>.
+
+ <li><p>Assert: <var>bucket</var> is an <a>abstract storage bucket</a>.
+
+ <li><p><a for=set>For each</a> <var>endpoint</var> of <a>registered storage endpoints</a> whose
+ <a for="storage endpoint">types</a> <a for=set>contain</a> <var>type</var>, set <var>bucket</var>'s
+ <a for="storage bucket">map</a>[<var>endpoint</var>'s <a for="storage endpoint">identifier</a>] to
+ a new <a for="storage bucket">area</a>.
+
+ <li><p>Return <var>bucket</var>.
+</ol>
+
+
+<h3 id=storage-proxy-maps>Storage proxy maps</h3>
+
+<p>A <dfn>storage proxy map</dfn> is equivalent to a <a for=/>map</a>, except that all operations
+are instead performed on its <dfn for="storage proxy map">backing map</dfn>.
+
+<p class="note">This allows for the <a for="storage proxy map">backing map</a> to be replaced.
+
+<hr>
+
+<p>To <dfn>obtain an abstract storage bucket area map</dfn>, given a <a>storage type</a>
+<var>type</var>, <a for=/>environment settings object</a> <var>environment</var>, and
+<a>storage identifier</a> <var>identifier</var>, run these steps:</p>
+
+<ol>
+ <li><p>Let <var>map</var> be null.
+
+ <li><p>If <var>type</var> is "<code>storage</code>", then set <var>map</var> to the user agent's
+ <a for="user agent">storage map</a>.
+
+ <li>
+  <p>Otherwise, if <var>type</var> is "<code>session-storage</code>", then set <var>map</var> to
+  <var>environment</var>'s <span class=XXX>browsing session</span>'s
+  <a for="browsing session">storage map</a>.
+
+  <p class="XXX">See
+  <a href="https://github.com/whatwg/html/issues/4782">whatwg/html issue #4782</a> and
+  <a href="https://github.com/whatwg/html/issues/5350">whatwg/html issue #5350</a> for defining
+  browsing session. It is roughly analogous to <a>top-level browsing context</a> except that it
+  cannot be replaced due to <code>Cross-Origin-Opener-Policy</code> or navigation.
+
+ <li><p>Assert: <var>map</var> is a <a for=/>storage map</a>.
+
+ <li><p>Let <var>unit</var> be the result of running <a>obtain a storage unit</a>, with
+ <var>map</var>, <var>environment</var>, and <var>type</var>.
+
+ <li><p>If <var>unit</var> is failure, then return failure.
+
+ <li><p>Let <var>bucket</var> be <var>unit</var>'s
+ <a for="storage unit">map</a>["<code>default</code>"].
+
+ <li><p>Let <var>area</var> be <var>bucket</var>'s
+ <a for="storage bucket">map</a>[<var>identifier</var>].
+
+ <li><p>Let <var>proxyMap</var> be a new <a>storage proxy map</a> whose
+ <a for="storage proxy map">backing map</a> is <var>area</var>'s
+ <a for="storage bucket/area">map</a>.
+
+ <li><p>Append a reference to <var>proxyMap</var> to <var>area</var>'s
+ <a for="storage bucket/area">proxy map reference set</a>.
+
+ <li><p>Return <var>proxyMap</var>.
+</ol>
+
+<p>To <dfn export>obtain a storage bucket area map</dfn>, given an
```

Comment:
The above-mentioned intro/model section I suggested should prominently point out that these are the entry points specifications will usually use.